### PR TITLE
MEC1501 documentation improvement and fix for UART2 console

### PR DESCRIPTION
**Pull request by franciscomunoz**
Update to the documentation(issue #17483). In addition, now the debug control register is forced to an specific value which will ensure that people booting from SPI can see the UART console.

### CODE REVIEW

**Comment by nashif** [comment]
@franciscomunoz  can you please set the title to something that really describes the content of this PR?

File: boards/arm/mec15xxevb_assy6853/doc/index.rst [hunk 1]
```diff
@@ -93,14 +101,14 @@ board. Advanced users may deviate from this recommendation.
 Jumper setting for MEC15xx EVB Assy 6853 Rev A1p0
 =================================================
 
-Power-related jumpers.
+Power-related jumpers
 ----------------------
```
Comment (ulfalizer) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Could shorten the separator too.

File: boards/arm/mec15xxevb_assy6853/doc/index.rst [hunk 2]
```diff
@@ -93,14 +101,14 @@ board. Advanced users may deviate from this recommendation.
...
 
-Power from +5V power brick connected to barrel connector P11 (5.5mm OD, 2.1mm ID)
-JP88 5-6 only
+Power from +5V power brick connected to barrel connector ``P11`` (5.5mm OD, 2.1mm ID)
+``JP88 5-6`` only.
```
Comment (ulfalizer) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Might help to use complete sentences, e.g.

    If ... of jumper ... is ..., then ... is connected to ...

File: boards/arm/mec15xxevb_assy6853/doc/index.rst [hunk 3]
```diff
@@ -141,34 +149,34 @@ These jumpers configure MEC1501 Boot-ROM straps.
 | 2-3         | 1-2        | 2-3          | 1-2         |
 +-------------+------------+--------------+-------------+
 
-JP96 1-2 pulls SHD SPI CS0# up to VTR2
+``JP96 1-2`` pulls SHD SPI CS0# up to VTR2
```
Comment (ulfalizer) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Could put a period at the end of the line if the sentence ends after "VTR2".

File: boards/arm/mec15xxevb_assy6853/doc/index.rst [hunk 4]
```diff
@@ -204,23 +212,34 @@ Jumper settings for MEC1503 144WFBGA Socket DC Assy 6856 Rev B1p0
...
-using a single physical connection.  In addition, it comes with sockets for
-JTAG only sessions.
+Building
+==========
+
+#. Build :ref:`hello_world` application as you would normally do.
+
+#. Once you obtain ``zephyr.bin``, proceed to use the microchip tool `SPI Image Gen`_
+   in order to create the the final binary. The output generated by this tool is
+   what you need to flash in the SHD SPI memory.
```
Comment (ulfalizer) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: Could simplify the language a bit:

    Once you have ``zephyr.bin``, use the SPI Image Gen microchip tool to create the final binary.
    You need the output from this tool to flash ...

**Review by tbursztyka — APPROVED** [review]

File: boards/arm/mec15xxevb_assy6853/pinmux.c [hunk 5]
```diff
@@ -44,7 +44,7 @@ static int board_pinmux_init(struct device *dev)
 	 * controlled by their respective PCR register (UART2).
 	 * For more details see table 44-1
 	 */
-	ECS_REGS->DEBUG_CTRL |= 0x4;
+	ECS_REGS->DEBUG_CTRL = 0x5;
```
Comment (albertofloyd) [inline boards/arm/mec15xxevb_assy6853/pinmux.c]: Use MCHP MCHP_ECS_DCTRL_DBG_EN  and others to be clear in the intention

File: boards/arm/mec15xxevb_assy6853/doc/index.rst (hunk 1 above)
Comment (franciscomunoz) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: done

File: boards/arm/mec15xxevb_assy6853/doc/index.rst (hunk 2 above)
Comment (franciscomunoz) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: reworded

File: boards/arm/mec15xxevb_assy6853/doc/index.rst (hunk 4 above)
Comment (franciscomunoz) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: done

File: boards/arm/mec15xxevb_assy6853/pinmux.c (hunk 5 above)
Comment (franciscomunoz) [reply boards/arm/mec15xxevb_assy6853/pinmux.c]: done

File: boards/arm/mec15xxevb_assy6853/doc/index.rst [hunk 6]
```diff
@@ -240,33 +261,21 @@ Flashing
...
-   in your spi nor.
-
-   .. zephyr-app-commands::
-      :zephyr-app: samples/hello_world
-      :board: mec15xxevb_assy6853
-      :goals: build flash
+   UART2 port and apply power.
 
-   You should see "Hello World! mec15xxevb_assy6853" in your terminal.
+    You should see ``"Hello World! mec15xxevb_assy6853"`` in your terminal.
```
Comment (dbkinder) [inline boards/arm/mec15xxevb_assy6853/doc/index.rst]: The ``Y`` in ``You`` should be indented the same as the ``U`` in ``UART2`` on the line above.  (The extra space indent changes how the output looks.)

**Review by dbkinder — CHANGES_REQUESTED** [review]
small tweak, otherwise Looks OK.

File: boards/arm/mec15xxevb_assy6853/doc/index.rst (hunk 6 above)
Comment (franciscomunoz) [reply boards/arm/mec15xxevb_assy6853/doc/index.rst]: done

**Review by dbkinder — APPROVED** [review]
Looks like @ulfalizer caught issues, so docs LGTM.  Thanks!

**Comment by franciscomunoz** [comment]
@galak Anything else to address? This is a trivial change.

File: ext/hal/microchip/mec/mec1501/component/ecs.h [hunk 7]
```diff
@@ -71,6 +71,8 @@
 #define MCHP_ECS_DCTRL_MODE_MASK0	0x03ul
 #define MCHP_ECS_DCTRL_MODE_MASK \
 	((MCHP_ECS_DCTRL_DBG_MODE_MASK0) << (MCHP_ECS_DCTRL_DBG_MODE_POS))
+
+#define MCHP_ECS_DCTRL_DBG_MODE_POS	1u
```
Comment (ioannisg) [inline ext/hal/microchip/mec/mec1501/component/ecs.h]: Where is this used ? Couldn't find.

Comment (ioannisg) [reply ext/hal/microchip/mec/mec1501/component/ecs.h]: Ah right, it was missing.

**Review by ioannisg — APPROVED** [review]

**Review by albertofloyd — APPROVED** [review]

**Pull request merged**